### ISSUE TITLE
question about instruction tuning data

### ISSUE THREAD

**Issue by mine-sweeper** [OptimalScale/LMFlow#196]
Hello, I have a question about the implementation of data loader.
in src/lmflow/models/hf_decoder_model.py there is a line
`text_column_name = "text" if "text" in column_names else column_names[0]`
where only "input" field is used, while "output" field is ignored.
in example/finetune.py there is a line
`lm_dataset = finetuner.group_text(`
where all training examples are concatenated and then reshaped to (xx, block_size), each row of which is regarded as a new example.
And there is another line
`result["labels"] = result["input_ids"].copy()`
In this way all tokens would be predicted.
It seems these tricks are used in pre-training stage.
So is instruction tuning stage implemented by "continue pre-training with instruction data" in current version?

**Comment by mine-sweeper** [comment]
https://github.com/OptimalScale/LMFlow/issues/96

**Comment by mine-sweeper** [comment]
Thanks for your reply.
So the results on the medical QA datasets are obtained by `text_only`? That's interesting, for I never expected such implementation.
Currently I modified the code a bit for `text2text` training, since I prefer not to compute loss on instruction tokens.
Waiting for your official implementation of `text2text`.

**Comment by shizhediao** [comment]
Hi,
The `text2text` data format has been well supported!
Please try it out! Thank you.
